doc/api/crypto.md

```diff
@@ -1886,10 +1854,15 @@ added: v0.1.92
 * `options` {Object} [`stream.Writable` options][]
 * Returns: {Sign}
 
-Creates and returns a `Sign` object that uses the given `algorithm`.
-Use [`crypto.getHashes()`][] to obtain an array of names of the available
-signing algorithms. Optional `options` argument controls the
-`stream.Writable` behavior.
+Creates and returns a `Sign` object that uses the given `algorithm`.  Use
+[`crypto.getHashes()`][] to obtain the names of the available digest algorithms.
+Optional `options` argument controls the `stream.Writable` behavior.
+
+In some cases, a `Sign` instance can be created using the name of a signature
+algorithm, such as 'RSA-SHA256', instead of a digest algorithm. This will use
+the corresponding digest algorithm. This does not work for all signature
+algorithms, such as 'ecdsa-with-SHA256', so it is best to always use digest
+algorithm names.
```

Comment:
I would probably express this in an even stronger way, as far as I know, this is specific to OpenSSL.

Reply:
I'm just moving the existing text closer to the API it refers to, I don't want to rework too much unless its got a serious problem. In this case, we never build with anything other than OpenSSL, so doesn't all the docs apply to OpenSSL? I'll add some specific text if you suggest, but I'm not sure what else to say, myself.